presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -872,6 +914,8 @@ private SubPlan buildSubPlan(PlanNode node, FragmentProperties properties, Rewri
         private Optional<PartitioningHandle> partitioningHandle = Optional.empty();
         private final Set<PlanNodeId> partitionedSources = new HashSet<>();
 
+        private Deque<Boolean> replicatedReadsQueue = new ArrayDeque<>();
```

Comment:
Do we really need a queue of booleans here? Why not have the join nodes pass in one boolean value via the context while walking the plan tree?

Reply:
The flag on the join node needs to queue in the context so that the children's tablescan nodes can pick up the flag for setting the correct distribution properties.